src/routes/quickstarts/react-native/react-native-testing-library.mdx

```diff
@@ -0,0 +1,242 @@
+
+Out of the box, React Native does a good job to polyfill browser API's that we might need in libraries that were originally built to run on the browser. Our problem when using Mirage in a Jest testing environment is that we don't have access to some properties that the browser provides by default. To get around this issue we must update Jest's environment immediately after its setup is finished.
+
+To update your Jest environment follow these steps:
+
+1. Create a setup file in the root of your project. For the purpose of this guide we'll call it `jest.setup.js`.
+
+2. Install a polyfill for XMLHttpRequest:
+
+```bash
+# Using npm
+npm install --save-dev xmlhttprequest
+
+# Using Yarn
+yarn add --dev xmlhttprequest
+```
+
+3. Patch the variable `global` in your `jest.setup.js` to add support for XMLHttpRequest, self and window.
+
+```js
+// jest.setup.js
+global.self = global;
+global.window = {};
+global.XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+```
+
+4. Tell Jest to load the environment that we've created by adding the [`setupFilesAfterEnv`](https://jestjs.io/docs/en/configuration.html#setupfilesafterenv-array) to your Jest configuration. We must add `jest.setup.js` as an item in the list of files that Jest must execute after setting up its environment.
+
+> Your jest configuration can be found on your package.json or in the root of your project with the name `jest.config.js`.
+
+```js
+// jest.config.js
+const jestPreset = require('@testing-library/react-native/jest-preset');
+
+module.exports = {
+  preset: '@testing-library/react-native',
+  setupFiles: [...jestPreset.setupFiles],
+  setupFilesAfterEnv: ['./jest.setup.js'],
+};
+```
+
+## Step 3: Define your server
+
+Create a new `server.js` file and define your mock server.
+
+Here's a basic example:
+
+```js
+// src/server.js
+import { Server, Model } from "miragejs"
+
+export function makeServer({ environment = "development", trackRequests = false } = {}) {
+  let server = new Server({
+    environment,
+    trackRequests,
+
+    models: {
+      user: Model,
+    },
+
+    seeds(server) {
+      server.create("user", { name: "Bob" })
+      server.create("user", { name: "Alice" })
+    },
+
+    routes() {
+      this.namespace = "api"
+
+      this.get("/users", schema => {
+        return schema.users.all()
+      })
+    },
+  })
+
+  return server
+}
+```
+
+> In a React Native App, put this file in `src/mirage/server.js` so that changes to it trigger rebuilds.
+
+## Step 4: Create a test file that uses Mirage
+
+Here's the component we'll be testing.
+
+```jsx
+// src/App.js
+import React, {useEffect, useState} from 'react';
+import {SafeAreaView, View, Text} from 'react-native';
+
+const App = () => {
+  let [users, setUsers] = useState([]);
+  let [serverError, setServerError] = useState();
+
+  useEffect(() => {
+    let fetchUsers = async () => {
+      try {
+        let res = await fetch('http://localhost:3000/v1/api/users');
```

Comment:
Does react native require the full domain name (localhost:3000) here? Is it possible to use relative paths.

Reply:
Nop, a full domain is not necessary. Updated the snippet to use relative paths instead.